lib/ui/painting.dart

```diff
@@ -4960,6 +4960,22 @@ class ImmutableBuffer extends NativeFieldWrapperClass2 {
   }
   void _init(Uint8List list, _Callback<void> callback) native 'ImmutableBuffer_init';
 
+  /// Create a buffer from the asset with key [assetKey].
+  ///
+  /// Returns `null` if the asset does not exist.
+  static ImmutableBuffer? fromAsset(String assetKey) {
+    final ImmutableBuffer buffer = ImmutableBuffer._(0);
+    bool success = false;
+    _fromAsset(assetKey, buffer, (void result) {
+      success = true;
```

Comment:
This is racy - see below. We want to use `_futurize` for this, or make our own completer.

Reply:
is this actually racy? I'm not really familiar with the native interop but it appears to be fully synchronous to me

Reply:
If someone sets the return value in the C++ side before calling the completer here I believe it will have unexpected results.